gnosis_scan_api.py

```diff
@@ -0,0 +1,81 @@
+import requests
+import json
+import datetime
+import time
+from constants import header, REQUEST_TIMEOUT, SUCCESS_CODE
+
+
+def get_block_range(year, month, day):
+    date_time = datetime.datetime(year, month, day, 0, 0)
+    start_timestamp = int(time.mktime(date_time.timetuple())) + 3 * 60 * 60
+    url = (
+        "https://api.gnosisscan.io/api?module=block&action=getblocknobytime"
+        + "&timestamp="
+        + str(start_timestamp)
+        + "&closest=after&apikey=YourApiKeyToken"
+    )
+    try:
+        response = requests.get(
+            url,
+            headers=header,
+            timeout=REQUEST_TIMEOUT,
+        )
+        if response.status_code == SUCCESS_CODE:
+            data = json.loads(response.text)
+            start_block = int(data["result"])
+    except Exception as e:
+        print(e)
+        return -1, -1
+
+    end_timestamp = start_timestamp + 7 * 24 * 60 * 60 
+    url = (
+        "https://api.gnosisscan.io/api?module=block&action=getblocknobytime"
+        + "&timestamp="
+        + str(end_timestamp)
```

Comment:
What happens if `end_timestamp` is equal to the time stamp of a block? If the block number of that block is returned, we might have the double counting issue from the other old query.

One ugly fix woud be
```suggestion
        + str(end_timestamp - 1)
```

Reply:
Good catch. Applied your fix